lib/compress.js

```diff
@@ -1731,6 +1731,7 @@ merge(Compressor.prototype, {
                 var val = {};
                 for (var i = 0, len = this.properties.length; i < len; i++) {
                     var prop = this.properties[i];
+                    if (prop instanceof AST_Expansion) return this;
```

Comment:
> but wouldn't that suppress optimisation of ({a:b}).a?

That expression behaves the same with or without the patch.

> expr.may_throw_on_access() asks if expr.foo would be safe to eliminate; your patch propagates that from ({a:b}).foo to b.foo, which seems a bit off.

I don't think that's the case. The patch is getting the element itself to decide if it'll throw.

But perhaps I am misunderstanding the point of `may_throw_on_access`. Would you be able to put together a fix for both array and object spreads on harmony?